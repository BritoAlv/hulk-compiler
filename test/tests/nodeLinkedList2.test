type Node(value : number) 
{
    value = value;
    previous = while(false) 0;
    getPrevious() : Node => self.previous;
    setPrevious(node : Node) : Node => self.previous := node;
    getValue() : number => self.value;
}

let a = new Node(10) in 
{
    a.setPrevious(new Node(20));
    print(a.getValue());
    print(a.getPrevious().getValue());
    a.getPrevious().getPrevious().getValue(); # Null reference error
};