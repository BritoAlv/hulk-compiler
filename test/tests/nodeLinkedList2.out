10.00000000
20.00000000
Null reference error

